fundamentals/android/auth-anonymous/Notes/app/src/main/java/com/notes/app/screens/sign_up/CredentialsExt.kt

```diff
@@ -0,0 +1,23 @@
+package com.notes.app.screens.sign_up
+
+import android.util.Patterns
+import java.util.regex.Pattern
+
+// Passwords should have at least six digits and include
+// one digit, one lower case letter and one upper case letter.
+private const val MIN_PASS_LENGTH = 6
+private const val PASS_PATTERN = "^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=\\S+$).{4,}$"
+
+fun String.isValidEmail(): Boolean {
+  return this.isNotBlank() && Patterns.EMAIL_ADDRESS.matcher(this).matches()
+}
+
+fun String.isValidPassword(): Boolean {
+  return this.isNotBlank() &&
+    this.length >= MIN_PASS_LENGTH &&
+    Pattern.compile(PASS_PATTERN).matcher(this).matches()
+}
+
+fun String.passwordMatches(repeated: String): Boolean {
+  return this == repeated
+}
```

Comment:
Do we really need this extension function? Seems like overkill for such trivial functionality (equals)

Reply:
Removed it, thanks!